base/replicator.go

```diff
@@ -245,13 +245,27 @@ func (r *Replicator) populateActiveTaskFromReplication(replication sgreplicate.S
 }
 
 func (r *Replicator) StopReplications() error {
-	 r.lock.Lock()
-	 defer r.lock.Unlock()
-	for replicationId, replication := range r.replications {
+
+	replicationIds := r.GetReplicationIds()
+
```

Comment:
If a replication is stopped by another goroutine here (after you get the ids, but before you call stopReplication), then stopReplication is going to fail and the rest of the replications won't be stopped.  This code would be more robust if it ignored 'not found' errors returned by stopReplication (as it would be safe to assume the replication had already been stopped).

Reply:
Agreed .. changed to warning and pushed up a new commit.